### PR TITLE
feat: Deployment protection for 'QA' and 'Production' workspaces

### PR DESCRIPTION
**Pull request by askayastha22**
Protect against accidentally deploying into workspaces like QA and Production from your local development environment.

## Description
Utilizes `pre-app-deploy` App Builder hook for the implementation.

Reference: https://developer.adobe.com/app-builder/docs/guides/app-hooks/

## Motivation and Context
During my onboarding session with @lenaorobei , @slitviachenko and @irenelagno, they pointed out how someone could accidentally deploy WIP features to the non-development workspaces.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by slitviachenko — APPROVED** [review]
LGTM, but I'm not sure we need to introduce a new dependency (`chalk`) just to display a message.

**Comment by askayastha22** [comment]
@slitviachenko Colors are nice. 🙂

File: .github/workflows/deploy_prod.yml [hunk 1]
```diff
@@ -33,6 +33,7 @@ jobs:
           EXPRESS_SDK_URL: https://dev.cc-embed.adobe.com/sdk/1p/v4/4.1.14/CCEverywhere.js
           SPLUNK_HEC__HEC_TOKEN: ${{ secrets.SPLUNK_HEC__HEC_TOKEN }}
           AIO_LOG_LEVEL: info
+          IS_CICD: 1
```
Comment (vtsaplin) [inline .github/workflows/deploy_prod.yml:36]: You can use the GITHUB_ACTIONS variable to differentiate when the script is being run locally or by GitHub Actions.

**Comment by vtsaplin** [comment]
I'm not sure if we should completely prevent manual deployment to prod and qa just yet, especially when a quick fix is needed. 

Wdyt?

**Comment by askayastha22** [comment]
The main idea is not to completely prevent manual deployment but protect against accidental deployment in QA and Production. Anyone can easily disable the hooks without committing any changes.

**Review by vtsaplin — CHANGES_REQUESTED** [review]
Looks good to me! Just a few minor things:
- I'd use the existing env var `GITHUB_ACTIONS`
- Might also note the hook in the readme

File: app.config.yaml [hunk 2]
```diff
@@ -41,3 +41,5 @@ application:
             function: actions/csv/index.js
             web: true
             runtime: nodejs:18
+  hooks:
+    pre-app-build: ./hooks/pre-app-build.js
```
Comment (lenaorobei) [inline app.config.yaml:45]: Don't we want `pre-app-deploy`?

Comment (vtsaplin) [reply app.config.yaml:45]: > the user will have to wait for the build to complete before they will know that they are in a protected workspace.

That's okey. It should not happen often.

I'd agree with @lenaorobei `pre-app-deploy` would be a better chose cuz [POLA](https://en.wikipedia.org/wiki/Principle_of_least_astonishment)

**Review by lenaorobei — APPROVED** [review]

**Review by vtsaplin — APPROVED** [review]

**Pull request merged**